curriculum/challenges/english/25-front-end-development/quiz-javascript-fundamentals/66edcd875b0d91de1fbbb492.md

```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
+Merge arrays or objects
 
 #### --answer--
 
-Placeholder answer
+Extract properties from objects and elements from arrays
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is a valid function name in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+123function
 
 ---
 
-Placeholder distractor 2
+function-name
 
 ---
 
-Placeholder distractor 3
+function
 
 #### --answer--
 
-Placeholder answer
+functionName
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of these is NOT a good practice for naming variables and functions in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+Use camelCase notation
 
 ---
 
-Placeholder distractor 2
+Start variable names with a letter, $ or _
 
 ---
 
-Placeholder distractor 3
+Using meaningful names
 
 #### --answer--
 
-Placeholder answer
+Using reserved keywords as variable names
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the primary purpose of a linter in JavaScript development?
 
 #### --distractors--
 
-Placeholder distractor 1
+To format code automatically
 
 ---
 
-Placeholder distractor 2
+To optimize code for performance
 
 ---
 
-Placeholder distractor 3
+To create documentation for code
 
 #### --answer--
 
-Placeholder answer
+To detect potential errors and style issues in code
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is NOT a linter for JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+ESLint
 
 ---
 
-Placeholder distractor 2
+standardJS
 
 ---
 
-Placeholder distractor 3
+quick-lint-js
 
 #### --answer--
 
-Placeholder answer
+Babel
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of these describes accessibility
 
 #### --distractors--
 
-Placeholder distractor 1
+Making sure code is accessible to other developers
 
 ---
 
-Placeholder distractor 2
+Writing secure code
 
 ---
 
-Placeholder distractor 3
+Controlling who can access certain data
 
 #### --answer--
 
-Placeholder answer
+Making web interfaces usable by all, including people with disabilities
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Why is using the 'var' keyword considered bad practice in modern JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+It's slower than let and const
 
 ---
 
-Placeholder distractor 2
+It causes a syntax error
 
 ---
 
-Placeholder distractor 3
+It doesn't allow type checking
 
 #### --answer--
 
-Placeholder answer
+It has function scope instead of block scope, which can lead to unexpected behaviors
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In JavaScript, how is memory for variables typically allocated?
 
 #### --distractors--
 
-Placeholder distractor 1
+Manually by the developer
 
 ---
 
-Placeholder distractor 2
+By the browser
 
 ---
 
-Placeholder distractor 3
+By the server
 
 #### --answer--
 
-Placeholder answer
+Automatically by the garbage collector
```

Comment:
Not sure if this is right. I always thought garbage collector is not responsible for allocating as well.

Reply:
Correct. The garbage collector reclaims memory.